lib/rubycritic/commands/base.rb

```diff
@@ -0,0 +1,16 @@
+require "rubycritic/commands/status_reporter"
+
+module Rubycritic
+  module Command
+    class Base
+      def initialize(options)
+        @options = options
+        @status_reporter = Rubycritic::Command::StatusReporter.new(@options)
+      end
+
+      def execute
+        @status_reporter
```

Comment:
I had an uneasy feeling about this in the first review already and on second thought I don't think we should do this:

You're using the template pattern here where you overwrite `execute` in subclasses. The problem with the way do it right now is that you're creating an undocumented, obscure dependency here - corresponding methods in subclasses **must** call `super`, because otherwise the return type changes signifcantly. And not only that, they need to call `super` in a certain order as well.

Let's not do this, this will confuse ourselves in 4 weeks from now. How about just raising NotImplementedError here and then return "@status_report" in all `executes` below?